### PR TITLE
Create TextWithIcon component

### PR DESCRIPTION
**Pull request by ingraso**
# Description

Text is displayed with icons throughout the whole webapp. However, there is no component or common way of doing this. With the `<TextWithIcon />` component there is now a generalized component that can be used for this purpose. An example of usage can be seen here:
image

There should be no visual changes to the text with icons.

Also fixes some TS errors and moves the tooltip on `<EventDetail />` to be on the icon rather than the text.

# Result
**Before**
image

**After**
image

# Testing

- [x] I have thoroughly tested my changes.

Tested by clicking through where changes have been done.

---

Resolves ABA-293

### CODE REVIEW

File: app/routes/events/components/EventDetail/index.tsx [hunk 1]
```diff
@@ -211,7 +212,7 @@ export default class EventDetail extends Component<Props, State> {
     const color = colorForEvent(event.eventType);
 
     const onRegisterClick = event.following
-      ? () => unfollow(event.following, event.id)
+      ? () => unfollow(event.id, event.id)
       : () => follow(currentUser.id, event.id);
```
Comment (ollfkaih) [inline app/routes/events/components/EventDetail/index.tsx:216]: Types were wrong here, but these lines are correct (#3592)

**Review by ollfkaih — CHANGES_REQUESTED** [review]

**Review by LudvigHz — APPROVED** [review]
LGTM

**Review by ivarnakken — APPROVED** [review]
Nice to standardize stuff like this 💯 Just fix up in Olav's comment and we should be good here

**Review by ollfkaih — APPROVED** [review]
Looks good now!

**Pull request merged**